homeassistant/components/yamaha_musiccast/media_player.py

```diff
@@ -1,291 +1,371 @@
+        self.schedule_update_ha_state()
+
+    async def async_media_play(self):
+        """Send play command."""
+        if self._is_netusb:
+            await self.coordinator.musiccast.netusb_play()
+
+    async def async_media_pause(self):
+        """Send pause command."""
+        if self._is_netusb:
+            await self.coordinator.musiccast.netusb_pause()
+
+    async def async_media_stop(self):
+        """Send stop command."""
+        if self._is_netusb:
+            await self.coordinator.musiccast.netusb_pause()
+
+    async def async_set_shuffle(self, shuffle):
+        """Enable/disable shuffle mode."""
+        if self._is_netusb:
+            await self.coordinator.musiccast.netusb_shuffle(shuffle)
+
+    async def async_select_sound_mode(self, sound_mode):
+        """Select sound mode."""
+        await self.coordinator.musiccast.select_sound_mode(self._zone_id, sound_mode)
 
     @property
     def media_image_url(self):
-        """Image url of current playing media."""
-        return self.media_status.media_image_url if self.media_status else None
+        """Return the image url of current playing media."""
+        return self.coordinator.musiccast.media_image_url if self._is_netusb else ""
+
+    @property
+    def media_title(self):
+        """Return the title of current playing media."""
+        if self._is_netusb:
+            return self.coordinator.data.netusb_track
+        if self._is_tuner:
+            return self.coordinator.musiccast.tuner_media_title
+
+        return None
 
     @property
     def media_artist(self):
-        """Artist of current playing media, music track only."""
-        return self.media_status.media_artist if self.media_status else None
+        """Return the artist of current playing media (Music track only)."""
+        if self._is_netusb:
+            return self.coordinator.data.netusb_artist
+        if self._is_tuner:
+            return self.coordinator.musiccast.tuner_media_artist
+
+        return None
 
     @property
-    def media_album(self):
-        """Album of current playing media, music track only."""
-        return self.media_status.media_album if self.media_status else None
+    def media_album_name(self):
+        """Return the album of current playing media (Music track only)."""
+        return self.coordinator.data.netusb_album if self._is_netusb else None
 
     @property
     def media_track(self):
-        """Track number of current playing media, music track only."""
-        return self.media_status.media_track if self.media_status else None
+        """Return the track number of current media (Music track only)."""
+        return -1
 
     @property
-    def media_title(self):
-        """Title of current playing media."""
-        return self.media_status.media_title if self.media_status else None
+    def repeat(self):
+        """Return current repeat mode."""
+        return (
+            {
+                "off": REPEAT_MODE_OFF,
```

Comment:
Can we define a constant that is the same as `REPEAT_MODE_MAPPING` but inverted?